src/AspNet.Security.OAuth.Autodesk/AutodeskAuthenticationHandler.cs

```diff
@@ -50,6 +51,41 @@ public partial class AutodeskAuthenticationHandler : OAuthHandler<AutodeskAuthen
         return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
     }
 
+    protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+    {
+        var tokenRequestParameters = new Dictionary<string, string>
+        {
+            ["redirect_uri"] = context.RedirectUri,
+            ["code"] = context.Code,
+            ["grant_type"] = "authorization_code"
+        };
+
+        // PKCE https://tools.ietf.org/html/rfc7636#section-4.5, see BuildChallengeUrl
+        if (context.Properties.Items.TryGetValue(OAuthConstants.CodeVerifierKey, out var codeVerifier))
+        {
+            tokenRequestParameters[OAuthConstants.CodeVerifierKey] = codeVerifier!;
+            context.Properties.Items.Remove(OAuthConstants.CodeVerifierKey);
+        }
+
+        using var requestContent = new FormUrlEncodedContent(tokenRequestParameters!);
+        using var requestMessage = new HttpRequestMessage(HttpMethod.Post, Options.TokenEndpoint);
+        requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        var credentials = Convert.ToBase64String(Encoding.ASCII.GetBytes(
+            $"{Options.ClientId}:{Options.ClientSecret}"));
```

Comment:
Technically, the standard way of computing the `Authorization: Basic [...]` header requires formURL-encoding the client identifier and client secret individually before computing their base64 representation. That said, I see in [the Autodesk docs that they don't mention formURL-encoding](https://aps.autodesk.com/en/docs/oauth/v2/tutorials/get-2-legged-token/#step-1-convert-client-id-and-secret-to-base64-encoded-string) so their implementation is probably not 100% compliant. Let's keep it this way.